pkg/controllers/sidecar/identity/member.go

```diff
@@ -63,27 +66,43 @@ func Retrieve(ctx context.Context, name, namespace string, kubeclient kubernetes
 	}, nil
 }
 
-func (m *Member) GetSeeds(ctx context.Context, kubeClient kubernetes.Interface) ([]string, error) {
-	var services *corev1.ServiceList
-	var err error
+// GetSeed returns a single node contact point used for bootstrapping.
+// A single node is enough because bootstrap process will fail if any of the nodes in cluster is down.
+func (m *Member) GetSeed(ctx context.Context, coreClient v1.CoreV1Interface) (string, error) {
+	pods, err := coreClient.Pods(m.Namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{
+			naming.ClusterNameLabel: m.Cluster,
+		}).String(),
+	})
+	if err != nil {
+		return "", err
+	}
 
-	sel := fmt.Sprintf("%s,%s=%s", naming.SeedLabel, naming.ClusterNameLabel, m.Cluster)
+	// Return static IP of itself to bootstrap first node.
+	if len(pods.Items) == 1 && pods.Items[0].Name == m.Name {
+		return m.StaticIP, nil
+	}
 
-	const maxRetryCount = 5
-	for retryCount := 0; ; retryCount++ {
-		services, err = kubeClient.CoreV1().Services(m.Namespace).List(ctx, metav1.ListOptions{LabelSelector: sel})
-		if err == nil && len(services.Items) > 0 {
-			break
-		}
-		if retryCount > 5 {
-			return nil, errors.New(fmt.Sprintf("failed to get seeds, error: %+v, len(services): %d", err, len(services.Items)))
+	// If at least one pod is ready, return the IP of service.
+	for _, pod := range pods.Items {
+		if podutils.IsPodReady(&pod) && pod.Name != m.Name {
+			svc, err := coreClient.Services(m.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
+			if err != nil {
+				return "", err
+			}
+			return svc.Spec.ClusterIP, nil
 		}
-		time.Sleep(1000 * time.Millisecond)
 	}
 
-	seeds := []string{}
-	for _, svc := range services.Items {
-		seeds = append(seeds, svc.Spec.ClusterIP)
+	// When all nodes are down, return service IP of the first created pod.
+	sort.Slice(pods.Items, func(i, j int) bool {
```

Comment:
this can return a service for the same pod, right?

Reply:
Yes. Although i don't think it matters. New pods are not scheduled when any of the previous nodes is down. And when node is already boostraped, seed list doesn't matter.
@asias Could you confirm the latter?